internal/api/handlers/middleware/middleware.go

```diff
@@ -0,0 +1,63 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type HeaderParam = string
+
+const (
+	ContentLength HeaderParam = "Content-Length"
+	UserAgent     HeaderParam = "User-Agent"
+	Host          HeaderParam = "Host"
+	ContentType   HeaderParam = "Content-Type"
+)
+
+// InjectedLogging uses logging middleware
+func InjectedLogging(logger *zap.Logger) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			defer func() {
+				if err := recover(); err != nil {
+					w.WriteHeader(http.StatusInternalServerError)
+					logger.Error("Failure occurred during request processing")
+				}
+			}()
+
+			userAgent := r.Header.Get(UserAgent)
+			if len(userAgent) == 0 {
+				userAgent = "-"
+			}
+
+			contentLength := r.Header.Get(ContentLength)
+			if len(contentLength) == 0 {
+				contentLength = "-"
+			}
+
+			host := r.Header.Get(Host)
+			if host == "" {
+				host = r.RemoteAddr
+			}
+
+			contentType := r.Header.Get(ContentType)
+			if contentType == "" {
+				contentType = "-"
+			}
+
```

Comment:
Do we wanna reject the request if these vars aren't set ? 

Sets a bit of min requirements to set the data correctly with the request.

Reply:
Eh I mean it couldn't hurt, we may however wanna prioritize validation checks for the request body content first